gqlalchemy/query_builders/memgraph_query_builder.py

```diff
@@ -79,8 +80,97 @@ def load_csv(self, path: str, header: bool, row: str) -> "DeclarativeBase":
 
         return self
 
+    def construct_subgraph_query(
+        self, node_label: Optional[str] = None, relationship_types: Optional[Union[str, Iterable[str]]] = None
+    ) -> str:
+        """Constructs a MATCH query defining a subgraph using zero or one node labels and relationship types.
+
+        Args:
+            node_label: A string representing a node label to be used in this subgraph.
+            relationship_types: A string or iterable of types of relationships used in the subgraph.
+
+        Returns:
+            a string representing a MATCH query for a path with given node labels and relationship types.
+        """
+        node = CypherNode(node_label)
+
+        # directed is OK only if we have single label, homogenous graphs (current implementation)
+        query = f" MATCH p={node}{CypherRelationship(relationship_types, RelationshipDirection.RIGHT)}{node}"
+
+        return query
+
+    def call(
+        self,
+        procedure: str,
+        arguments: Optional[Union[str, Tuple[Union[str, int, float]]]] = None,
+        node_label: Optional[str] = None,
+        relationship_types: Optional[Union[str, Iterable[str]]] = None,
+        subgraph_query: str = None,
+    ) -> "DeclarativeBase":
+        """Override of base class method to support Memgraph's subgraph functionality.
+
+        Method can be called with node labels and relationship types, both being optional, which are used to construct
+        a subgraph, or if neither is provided, a subgraph query is used, which can be passed as a string representing a
+        Cypher query defining the MATCH clause which selects the nodes and relationships to use.
+
+        Args:
+            procedure: A string representing the name of the procedure in the
+              format `query_module.procedure`.
+            arguments: A string representing the arguments of the procedure in
+              text format.
+            node_label: Label of nodes to be used in the subgraph.
+            relationship_types: Types of relationships to be used in the subgraph.
+            subgraph_query: Optional way to define the subgraph via a Cypher MATCH clause.
+
+        Returns:
+            A `DeclarativeBase` instance for constructing queries.
+
+        Examples:
+            Python: `call('export_util.json', '/home/user', "LABEL", ["TYPE1", "TYPE2"]).execute()
+            Cypher: `MATCH p=(:LABEL)-[:TYPE1 | :TYPE2]->(:LABEL) WITH project(p) AS graph
+                    CALL export_util.json(graph, '/home/user')`
+
+            or
+
+            Python: `call('export_util.json', '/home/user', subgraph_query="MATCH (:LABEL)-[:TYPE]->(:LABEL)").execute()
+            Cypher: `MATCH p=(:LABEL)-[:TYPE1]->(:LABEL) WITH project(p) AS graph
+                    CALL export_util.json(graph, '/home/user')`
+        """
```

Comment:
This subgraph_query is not a path

Reply:
this is intentional, `ProjectPartialQuery` checks if there is `p=` (it can be anything instead of p), and if not, adds it, so you can simply write a query you want to define your subgraph, no need to know about our specific implementation, which uses paths.